src/app/launcher/create-app/dependency-editor-step/dependency-editor-step.component.ts

```diff
@@ -0,0 +1,200 @@
+    Host,
+    Input,
+    KeyValueDiffers,
+    OnInit,
+    OnDestroy,
+    ViewEncapsulation
+} from '@angular/core';
+import { FormsModule } from '@angular/forms';
+import { Subscription } from 'rxjs/Subscription';
+
+import { DependencyEditorService } from '../../service/dependency-editor.service';
+import { DependencyCheckService } from '../../service/dependency-check.service';
+import { Selection } from '../../model/selection.model';
+import { LauncherComponent } from '../../launcher.component';
+import { LauncherStep } from '../../launcher-step';
+import { DependencyEditor } from '../../model/dependency-editor/dependency-editor.model';
+import { Summary } from '../../model/summary.model';
+
+@Component({
+    encapsulation: ViewEncapsulation.None,
+    selector: 'f8launcher-dependencychecker-createapp-step',
+    templateUrl: './dependency-editor-step.component.html',
+    styleUrls: ['./dependency-editor-step.component.less']
+})
+export class DependencyEditorCreateappStepComponent extends LauncherStep implements OnInit, OnDestroy, DoCheck {
+    @Input() id: string;
+
+    public github: string = '';
+    public gitref: string = '';
+    public boosterInfo: any = null;
+    public metadataInfo: any = null;
+    private cacheInfo: any = {};
+    private changes: any = {};
+
+    private subscriptions: Subscription[] = [];
+    constructor(
+        @Host() public launcherComponent: LauncherComponent,
+        private depEditorService: DependencyEditorService,
+        private dependencyCheckService: DependencyCheckService,
+        private keyValueDiffers: KeyValueDiffers
+    ) {
+        super();
+        if (this.launcherComponent.summary) {
+            this.launcherComponent.summary['dependencyEditor'] = new DependencyEditor();
+        }
+    }
+
+    ngOnDestroy() {
+        this.subscriptions.forEach((sub) => {
+            sub.unsubscribe();
+        });
+    }
+
+
+    ngOnInit() {
+        this.changes = this.keyValueDiffers.find(this.launcherComponent.summary).create(null);
+        this.launcherComponent.addStep(this);
+        this.dependencyCheckService.getDependencyCheck()
+        .subscribe((val) => {
+            this.metadataInfo = val;
+        });
+    }
+
+    ngDoCheck() {
+        let updates: any = this.changes.diff(this.launcherComponent.summary);
+        if (updates) {
+            updates.forEachChangedItem((r: any) => {
+                this.listenForChanges(r);
+            });
+            updates.forEachAddedItem((r: any) => {
+                this.listenForChanges(r);
+            });
+            updates.forEachRemovedItem((r: any) => {
+                this.listenForChanges(r);
+            });
+        }
+    }
+
+    // Accessors
+    /**
+     * Returns indicator that step is completed
+     *
+     * @returns {boolean} True if step is completed
+     */
+    get stepCompleted(): boolean {
+            return (this.launcherComponent.summary.dependencyEditor !== undefined);
+    }
+
+    /**
```

Comment:
there should not be a line break between the docs comments and the method declaration